ReactSkia/modules/platform/libcurl/RSkNetworkingModule.cpp

```diff
@@ -263,18 +293,15 @@ void RSkNetworkingModule::headerCallbackWrapper(NetworkRequest *networkRequest,
 }
 
 void RSkNetworkingModule::writeMemoryCallbackWrapper(NetworkRequest *networkRequest, char* writeMemoryBuffer, size_t realSize) {              
-  if(networkRequest->responseBuffer_ == nullptr && networkRequest->contentSize_ != 0)
-    networkRequest->responseBuffer_  = (char *) malloc(networkRequest->contentSize_+1);
+  networkRequest->responseBuffer_  = (char *) realloc(networkRequest->responseBuffer_ , networkRequest->responseBufferOffset_+realSize+1);
   RNS_LOG_ASSERT((networkRequest->responseBuffer_), "responseBuffer cannot be null");
-  memcpy(&(networkRequest->responseBuffer_[networkRequest->responseBufferOffset_]), writeMemoryBuffer, realSize);
+  memcpy(&(networkRequest->responseBuffer_ [networkRequest->responseBufferOffset_]), writeMemoryBuffer, realSize);
   networkRequest->responseBufferOffset_ += realSize;
-  if(networkRequest->contentSize_ == networkRequest->responseBufferOffset_) {
-     networkRequest->responseBuffer_[networkRequest->responseBufferOffset_] = 0;
-  }
+  networkRequest->responseBuffer_[networkRequest->responseBufferOffset_] = 0;
```

Comment:
this has to be added when sendData ? because if we are getting data in chunks, then everytime we are appending '0' for the buffer